src/main/java/com/google/finapp/SpannerDaoJDBCImpl.java

```diff
@@ -0,0 +1,176 @@
+        PreparedStatement ps =
+            connection.prepareStatement(
+                "INSERT INTO Customer\n"
+                    + "(CustomerId, Name, Address)\n"
+                    + "VALUES\n"
+                    + "(?, ?, ?)")) {
+      ps.setBytes(1, customerId.toByteArray());
+      ps.setString(2, name);
+      ps.setString(3, address);
+      ps.executeUpdate();
+    } catch (SQLException e) {
+      throw new SpannerDaoException(e);
+    }
+  }
+
+  public void createAccount(
+      ByteArray accountId, AccountType accountType, AccountStatus accountStatus, BigDecimal balance)
+      throws SpannerDaoException {
+    try (Connection connection = DriverManager.getConnection(this.connectionUrl);
+        PreparedStatement ps =
+            connection.prepareStatement(
+                "INSERT INTO Account\n"
+                    + "(AccountId, AccountType, AccountStatus, Balance, CreationTimestamp)\n"
+                    + "VALUES\n"
+                    + "(?, ?, ?, ?, PENDING_COMMIT_TIMESTAMP())")) {
+      ps.setBytes(1, accountId.toByteArray());
+      ps.setInt(2, accountType.getNumber());
+      ps.setInt(3, accountStatus.getNumber());
+      ps.setBigDecimal(4, balance);
+      ps.executeUpdate();
+    } catch (SQLException e) {
+      throw new SpannerDaoException(e);
+    }
+  }
+
+  public void addAccountForCustomer(
+      ByteArray customerId, ByteArray accountId, ByteArray roleId, String roleName)
+      throws SpannerDaoException {
+    try (Connection connection = DriverManager.getConnection(this.connectionUrl);
+        PreparedStatement ps =
+            connection.prepareStatement(
+                "INSERT INTO CustomerRole\n"
+                    + "(CustomerId, AccountId, RoleId, Role)\n"
+                    + "VALUES\n"
+                    + "(?, ?, ?, ?)")) {
+      ps.setBytes(1, customerId.toByteArray());
+      ps.setBytes(2, accountId.toByteArray());
+      ps.setBytes(3, roleId.toByteArray());
+      ps.setString(4, roleName);
+      ps.executeUpdate();
+    } catch (SQLException e) {
+      throw new SpannerDaoException(e);
+    }
+  }
+
+  public void moveAccountBalance(ByteArray fromAccountId, ByteArray toAccountId, BigDecimal amount)
+      throws SpannerDaoException {
+    try (Connection connection = DriverManager.getConnection(this.connectionUrl);
+        PreparedStatement readStatement =
+            connection.prepareStatement(
+                "SELECT AccountId, Balance FROM Account WHERE (AccountId = ? or AccountId = ?)")) {
+      connection.setAutoCommit(false);
+      byte[] fromAccountIdArray = fromAccountId.toByteArray();
+      byte[] toAccountIdArray = toAccountId.toByteArray();
+      readStatement.setBytes(1, fromAccountIdArray);
+      readStatement.setBytes(2, toAccountIdArray);
+      java.sql.ResultSet resultSet = readStatement.executeQuery();
+      BigDecimal sourceAmount = new BigDecimal(Long.MIN_VALUE);
+      BigDecimal destAmount = new BigDecimal(Long.MIN_VALUE);
```

Comment:
I had to initialize the `BigDecimal`s here to resolve the compilation errors (since the following code doesn't guarantee that `sourceAmount` and `destAmount` will be assigned). See later comment.